.clang-format

```
@@ -0,0 +1,47 @@
+---
+### Run using:
+###     find ./src/libs/ascent/ -regex '.*\.\(c\|cpp\|h\|hpp\)' -exec clang-format -i {} +
+
+
+#############################
+### Alignment and Spacing ###
+#############################
+
+# Basic spacing and line length limits
+ColumnLimit: 79
+PenaltyExcessCharacter: 1000000 # Allows for strings to overflow past column
+AlignArrayOfStructures: Left    # Alignes columns of structures so they all line up clearly
+AlignConsecutiveAssignments: false # Don't bother lining up `=` symbols in consecutive lines
+AlignConsecutiveDeclarations: false # Don't bother lining up `=` symbols in consecutive lines
+
+# For lines that get wrapped, this forces one parameter per line and aligns them
+BinPackParameters: false        # One param per line if wrapped
+BinPackArguments: false         # One argument per line if wrapped
+AlignAfterOpenBracket: Align    # Align wrapped args/params under the first
+AllowAllParametersOfDeclarationOnNextLine: false
```

```suggestion
AllowAllParametersOfDeclarationOnNextLine: true
```
I think this should be true. Sometimes namespace::method() names get so long that it is useful to put parameter declarations on the following lines.

---

What I don't like is that when you make that change you get:

```
conduit::Node ASCENT_API field_reduction_max(
    const conduit::Node &field, const std::string &component = "");
```

rather than when it is false it gives:
```
conduit::Node ASCENT_API
history_gradient_range(const conduit::Node &y_values,
                       const conduit::Node &dx_values);
```

---

Does it let you put the arguments on multiple lines when set to `true`?

---

It does, but it seems to prioritize grouping things in a way I don't love. When there are enough arguments or the named are long enough both true and false will wrap the arguments onto their own lines